### PR TITLE
Fix `vault path-help` for selected paths with bad regexps

### PR DESCRIPTION
**Pull request by maxb**
**Describe the bug**
When a Vault API path has been defined with a regex that requires a final slash, the `vault path-help` command is unable to retrieve help for it.

**Underlying cause**
Vault's code automatically appends a trailing slash on to all paths used for `ListOperation`s, but removes a trailing slash for all other operations, including `HelpOperation`s. In order for a path pattern to match both `ListOperation` and `HelpOperation`, it must accept both the presence and absence of a trailing slash.

**To Reproduce, or test the fix**
1. Start a Vault dev server
2. Run `vault path-help auth/token/accessors/`
3. See an `unsupported path` error

**Expected behavior**
Help message displayed as expected.

**Fix**
Update the few path regex patterns that fall into this trap to end with `/?` instead of `/`.

Update a comment to explain why.

### CODE REVIEW

**Comment by maxb** [comment]
Hi @hsimon-hashicorp @miagilepner,

Apologies for the direct @-ing - I wasn't sure if there was any better contact method, so I just picked some people I have seen triage community submissions in the past.

I have quite a few open Vault PRs that have been sitting around for a while, so I picked this one - one of my smaller, hopefully easier to review ones - in which to ask: Is there anything I can do to make these easier for HashiCorp engineers to review and merge?

**Comment by hghaf099** [comment]
@maxb I will be reviewing this PR. Thanks for putting it together. It looks great from the first glance. I just need to confirm the changes to the sudo paths internally.

**Review by averche — APPROVED** [review]
This looks reasonable to me. I've tried out the steps to reproduce the issue and the fix and both work! Thanks again for contributing, @maxb!

I'll let @hghaf099 confirm the sudo path logic and merge the PR.

**Comment by maxb** [comment]
> @maxb I will be reviewing this PR. Thanks for putting it together. It looks great from the first glance. I just need to confirm the changes to the sudo paths internally.

@hghaf099 Please let me know if there is any evidence I can gather and add to this PR to assist with this process?

It would be great to get it in before its 6-month anniversary, and then I can move on to trying to attract a HashiCorp sponsor to some of my other open PRs.

**Review by hghaf099 — APPROVED** [review]

File: api/plugin_helpers.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ const (
 // path matches that path or not (useful specifically for the paths that
 // contain templated fields.)
 var sudoPaths = map[string]*regexp.Regexp{
-	"/auth/token/accessors/":                        regexp.MustCompile(`^/auth/token/accessors/?$`),
+	"/auth/token/accessors":                         regexp.MustCompile(`^/auth/token/accessors/?$`),
```
Comment (hghaf099) [inline api/plugin_helpers.go:43]: I think it is worth adding these modified paths to the `TestIsSudoPath` in [here](https://github.com/hashicorp/vault/blob/00295d2bf5b6cb67f2a939bd602b780aa63831cb/api/plugin_helpers_test.go#L8) just to keep track of them.

**Review by hghaf099 — COMMENTED** [review]
Looks great!

File: api/plugin_helpers.go (hunk 1 above)
Comment (maxb) [reply api/plugin_helpers.go:43]: I'm happy to add a couple more test cases focussed around list-only endpoints, but I'm not sure what you mean by "keep track of them" - the existing test cases only test some represenatative examples of sudo paths to confirm the code is correct, so I don't think I should copy/paste the same tests for `auth/token/accessors` and `sys/config/ui/headers`, which would be identical apart from the constant string.

Comment (hghaf099) [reply api/plugin_helpers.go:43]: It would be great to add some tests around list-only endpoints.

Comment (maxb) [reply api/plugin_helpers.go:43]: Done!

Comment (hghaf099) [reply api/plugin_helpers.go:43]: Great! Thank you!

**Pull request merged**